### PR TITLE
Created shell plugin for Tugboat.

### PR DESCRIPTION
**Pull request by markdorison**
Created shell plugin for [Tugboat CLI](https://docs.tugboatqa.com/tugboat-cli/).

### CODE REVIEW

**Review by SimonBarendse — APPROVED** [review]
Thank you Mark! This looks excellent

File: plugins/tugboat/access_token.go [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+					Length: 32,
+					Charset: schema.Charset{
+						Uppercase: true,
+						Lowercase: true,
+						Digits:    true,
+					},
+				},
+			},
+		},
+		DefaultProvisioner: provision.TempFile(tugboatConfig, provision.AtFixedPath("~/.tugboat.yml")),
```
Comment (florisvdg) [inline plugins/tugboat/access_token.go:35]: One downside of using the file provisioner with a fixed path is that it users won't be able to test drive this plugin without overwriting their existing configuration.

Seems that Tugboat also reads from the [`TUGBOAT_API_TOKEN` environment variable](https://github.com/TugboatQA/docker-tugboat-cli#read-the-token-using-an-environment-variable), so if that works well then that may be the more suitable option.

**Review by florisvdg — COMMENTED** [review]
Thanks for the contribution!

File: plugins/tugboat/access_token.go (hunk 1 above)
Comment (markdorison) [reply plugins/tugboat/access_token.go:35]: @florisvdg I was having trouble finding that documentation; thanks for sharing! Updating with your suggestion.

**Comment by DCKcode** [comment]
Thanks for your contribution 😄

**Review by DCKcode — APPROVED** [review]

**Comment by markdorison** [comment]
My pleasure; this was fun!

**Review by SimonBarendse — APPROVED** [review]

**Pull request merged**